rust/connlib/libs/tunnel/src/lib.rs

```diff
@@ -242,7 +229,14 @@ where
         registry = register_default_interceptors(registry, &mut media_engine)?;
         let mut setting_engine = SettingEngine::default();
         setting_engine.detach_data_channels();
-        // TODO: Enable UDPMultiplex (had some problems before)
+        setting_engine.set_ip_filter(Box::new({
+            let resources = Arc::clone(&resources);
+            move |ip| !resources.read().values().any(|res_ip| res_ip.contains(ip))
+        }));
+
+        setting_engine.set_interface_filter(Box::new({
+            |name| !name.contains("utun") && name != "tun-firezone"
```

Comment:
On Android unfortunately I don't think we have a deterministic interface name. I believe it's usually `ppp0` but we'd want to either make it deterministic or set it ourselves.

Reply:
Hmm retrieving the name from `IfaceDevice` would require to make this function slightly more complicated since at this point it doesn't exist, not hard but more hassle, the risk of using the tunnel interface is minimal so let's leave that for a follow up PR, #2163